### PR TITLE
Adapted to display moonbeam author

### PR DESCRIPTION
**Pull request by joelamouche**
replacement of https://github.com/polkadot-js/api/pull/3082/files

- swap Consensus-preRuntime check
- add non-aura non-babe author extraction

### CODE REVIEW

File: packages/api-derive/src/type/util.ts [hunk 1]
```diff
@@ -4,19 +4,19 @@
 import type { AccountId, Digest } from '@polkadot/types/interfaces';
 
 export function extractAuthor (digest: Digest, sessionValidators: AccountId[] = []): AccountId | undefined {
-  const [pitem] = digest.logs.filter(({ type }) => type === 'PreRuntime');
+  const [citem] = digest.logs.filter(({ type }) => type === 'Consensus');
 
   // extract from the substrate 2.0 PreRuntime digest
-  if (pitem) {
-    const [engine, data] = pitem.asPreRuntime;
+  if (citem) {
```
Comment (jacogr) [inline packages/api-derive/src/type/util.ts:10]: Small nitpick, we probably want to drop the comments on the previous line and on line 17 now.

File: packages/types/src/generic/ConsensusEngineId.ts [hunk 2]
```diff
@@ -104,6 +108,11 @@ export class GenericConsensusEngineId extends u32 {
       return this._getPowAuthor(bytes);
     }
 
+    // Moonbeam is neither Aura nor Babe nor Pow and uses h160 addresses
+    if (bytes.length === 20) {
+      return this._getH160Author(bytes);
```
Comment (jacogr) [inline packages/types/src/generic/ConsensusEngineId.ts:113]: +1

**Comment by joelamouche** [comment]
Is there a way to check codeclimate conditions with the cli before pushing?

**Comment by jacogr** [comment]
Ignore CC, it really is a guide rather than a requirement. In this case can break it down into move pieces but the code would then actually lose maintainability since you can follow it 100% atm. In a lot of cases it does help, in some cases (like this), the alternative is worse.

**Review by jacogr — APPROVED** [review]
Thank you!

**Pull request merged**